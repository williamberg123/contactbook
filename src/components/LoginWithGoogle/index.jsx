import PropTypes from 'prop-types';
import StyledLoginWithGoogle from './styles';

export default function LoginWithGoogle({ children, buttonFunc }) {
	return (
		<StyledLoginWithGoogle onClick={buttonFunc}>
			{children}
		</StyledLoginWithGoogle>
	);
}

LoginWithGoogle.propTypes = {
	children: PropTypes.node.isRequired,
	buttonFunc: PropTypes.func.isRequired,
};
