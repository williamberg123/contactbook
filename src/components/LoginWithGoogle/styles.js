import styled from 'styled-components';

const StyledLoginWithGoogle = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	padding: 15px;
	cursor: pointer;
	margin: 20px 0;
	background-color: #e5e5e5;
	color: #00bc5e;
	border-radius: 5px;
	font-weight: bold;

	&:hover {
		background-color: #d8d8d8;
	}

	& > svg {
		width: 25px;
		height: 25px;
		margin-left: 10px;
	}
`;

export default StyledLoginWithGoogle;
