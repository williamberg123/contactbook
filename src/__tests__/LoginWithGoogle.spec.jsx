import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import LoginWithGoogle from '../components/LoginWithGoogle';

describe('<LoginWithGoogle />', () => {
	it('should render LoginWithGoogle button correctly', () => {
		render(<LoginWithGoogle>Entrar com o google</LoginWithGoogle>);
		expect(screen.getByRole('button', { name: /entrar com o google/i })).toBeInTheDocument();
	});

	it('should call function when button is clicked', () => {
		const fn = vi.fn();
		render(<LoginWithGoogle buttonFunc={fn}>Entrar com o google</LoginWithGoogle>);

		fireEvent.click(screen.getByRole('button', { name: /entrar com o google/i }));
		expect(fn).toBeCalled();
	});

	it('should match snapshot', () => {
		const fn = vi.fn();
		render(<LoginWithGoogle buttonFunc={fn}>Entrar com o google</LoginWithGoogle>);

		expect(screen.getByRole('button', { name: /entrar com o google/i })).toMatchSnapshot();
	});
});
