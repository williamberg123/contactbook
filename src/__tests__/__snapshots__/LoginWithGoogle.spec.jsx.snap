// Vitest Snapshot v1

exports[`<LoginWithGoogle /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px;
  cursor: pointer;
  margin: 20px 0;
  background-color: #e5e5e5;
  color: #00bc5e;
  border-radius: 5px;
  border: none;
  font-weight: bold;
}

.c0:hover {
  background-color: #d8d8d8;
}

.c0 > svg {
  width: 25px;
  height: 25px;
  margin-left: 10px;
}

<button
  class="c0"
>
  Entrar com o google
</button>
`;
